### PR TITLE
[Dominion] Profile Page Hero

### PR DESCRIPTION
**Pull request by karimkawambwa**
## Description

Feature profile page hero.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Screenshots

Screen Shot 2019-03-12 at 11 21 20 AM
Screen Shot 2019-03-12 at 11 21 45 AM

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: dominion/dominion_ui/src/components/Hero/Hero.js [hunk 1]
```diff
@@ -34,14 +34,43 @@ const styles = theme => ({
...
+  detailLabel: {
+    color: '#8d8d8c',
+    fontFamily: 'Montserrat',
+    fontSize: '0.688em',
+    fontWeight: 500,
+    fontStyle: 'normal',
+    fontStretch: 'normal',
+    lineHeight: 2.09,
+    letterSpacing: 'normal'
+  },
```
Comment (kilemensi) [inline dominion/dominion_ui/src/components/Hero/Hero.js:70]: You're setting font explicitly here (on `detail`, `detailLabel`, etc.). These should come directly from the theme. If you want to override theming, say use `h3` component but `body1` style, then:
```jsx
<Typography variant="body1" component="h3">...</Typography>
```
should do the trick.

File: dominion/dominion_ui/src/components/Hero/Hero.js [hunk 2]
```diff
@@ -74,32 +103,48 @@ const styles = theme => ({
...
-function HeroTitleComponent({ classes, children }) {
+function HeroTitleComponent({ classes, children, breakWord, small }) {
   return (
-    <Typography variant="h1" className={classes.text}>
+    <Typography
+      variant="h1"
+      className={`${classes.title} ${
+        breakWord ? classes.titleWordBreak : null
+      } ${small ? classes.titleFontSmall : null}`}
+    >
```
Comment (kilemensi) [inline dominion/dominion_ui/src/components/Hero/Hero.js:143]: Looks like you want to have multiple classes here. If that's the case, we've already included [classnames](https://github.com/JedWatson/classnames), a small library to handle such cases elegantly.

File: dominion/dominion_ui/src/components/Hero/Hero.js [hunk 3]
```diff
@@ -119,6 +164,35 @@ HeroDescriptionComponent.propTypes = {
...
+  children: PropTypes.isRequired,
+  label: PropTypes.string.isRequired,
+  small: PropTypes.bool.isRequired
+};
+
+const HeroDetail = withStyles(styles)(HeroDetailComponent);
+
 function HeroButtonComponent({ classes, children, onClick }) {
   return (
     <Grid container sm={12} alignItems="center" style={{ paddingTop: '2rem' }}>
```
Comment (kilemensi) [inline dominion/dominion_ui/src/components/Hero/Hero.js]: `container` _shouldn't_ contain any of the `xs`, `sm`, etc., **unless** it's both a `container` and an `item` i.e. `<Grid item xs={12} container ...`

File: dominion/dominion_ui/src/components/Hero/ProfileHero.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+function ProfileHero({ classes }) {
+  const selectedCountry = window.selected_country || {};
+  return (
+    <Hero>
+      <HeroTitleGrid quater>
+        <HeroTitle breakWord small>
+          Eastern Cape
+        </HeroTitle>
+        <p style={{ color: '#8d8d8c', fontSize: '0.75em' }}>
```
Comment (kilemensi) [inline dominion/dominion_ui/src/components/Hero/ProfileHero.js]: Any reason why we're setting `style` like this? I've seen it being used in a couple of components already.

File: dominion/templates/_base.html [hunk 5]
```diff
@@ -11,6 +11,9 @@
 {% endblock %}
 
 {% block body_javascript %}
+  <script src="{% static asset_manifest.runtime %}"></script>
+  <script src="{% static asset_manifest.vendor %}"></script>
+  <script src="{% static asset_manifest.main %}"></script>
```
Comment (kilemensi) [inline dominion/templates/_base.html:16]: The reason I usually put my custom scripts **after** a call to `{{ block.super }}` is to make sure if there is any name clash, my scripts win out.

File: dominion/templates/country_page.html [hunk 6]
```diff
@@ -21,15 +21,12 @@
   {% include 'blocks/partners.html' %}
 {% endblock content_container %}
 
-{% block body_javascript_extra %}
+{% block homepage_javascripb_libs %}
```
Comment (kilemensi) [inline dominion/templates/country_page.html]: I think `homepage_javascripb_libs` is only used to _include_ libs and not write custom JS. That's why this code was put in the more general `body_javascript_extra`. Did it cause any errors?

**Review by kilemensi — CHANGES_REQUESTED** [review]
Looks good @karimkawambwa but few things missing:

 1. Values are hard-coded. To see variable names and how they're pulled from Django see: https://github.com/OpenUpSA/wazimap/blob/fde22a0874020cf0ae013aeec7ab55b7c5a70b27/wazimap/templates/profile/profile_detail.html#L27-L71
 2. There should be a small padding/margin on the left of hero (about 18px)
Screenshot from 2019-03-12 16-37-21
 3. The space between Main nav and hero is a bit on too much side (On my screen it's more than 100px when it should be around 65px)

File: dominion/templates/country_page.html (hunk 6 above)
Comment (karimkawambwa) [reply dominion/templates/country_page.html]: Khadija changed this due to map issues when it comes to drawing. @KhadijaMahanga might better explain this.

**Comment by karimkawambwa** [comment]
I think @KhadijaMahanga will address the issues since she is working on this branch.

**Comment by KhadijaMahanga** [comment]
@clemence @karimkawambwa this PR is ready for review

**Review by kilemensi — CHANGES_REQUESTED** [review]
Quick review @KhadijaMahanga, maps are _off_.
Screenshot from 2019-03-14 10-28-51

-------

Also in South Africa

File: dominion/dominion_ui/src/components/Hero/ProfileHero.js [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+    const geoid = window.geography.full_geoid;
+    const api = createAPI();
+    const level = await api.getGeoLevel(geoid);
+    this.setState({ level, geoid });
+  }
+
+  render() {
+    const { population } = window.population;
+    const { populationDensity } = window.population_density;
+    const { level, geoid } = this.state;
```
Comment (karimkawambwa) [inline dominion/dominion_ui/src/components/Hero/ProfileHero.js:82]: Can't we have one profile data object from window ?

File: dominion/dominion_ui/src/components/Hero/ProfileHero.js [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+            handleIconClick={null}
+            isComparisonSearch
+            placeholder="Compare this with"
+            thisGeoId={geoid}
+            icon={searchIcon}
+          />
+        </HeroTitleGrid>
+        <Grid id="slippy-map" className={classes.map} />
+        <Typography variant="body2" className={classes.release}>
+          {window.primary_releases} {window.primary_releases_year}
```
Comment (karimkawambwa) [inline dominion/dominion_ui/src/components/Hero/ProfileHero.js]: there is a `caption` that is concat `Census 2016`. I use it in releases section.

File: dominion/dominion_ui/src/components/ReleaseDropdown.js [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+              fontSize="large"
+              className={classes.KeyboardArrow}
+            />
+          )}
+        </Button>
+        {isDropdownOpen ? (
+          <MenuList className={classes.menuList}>
+            <MenuItem item className={classes.menuListItem}>
+              <a
+                href={`${window.location.href.replace(/\/$/, '')}?release=${
```
Comment (karimkawambwa) [inline dominion/dominion_ui/src/components/ReleaseDropdown.js]: you can just have it as `href={'?release=${year}'}`

File: dominion/dominion_ui/src/components/ReleaseDropdown.js [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+                  window.primary_releases_year
+                }`}
+                className={classes.link}
+              >
+                {window.primary_releases} {'  '} {window.primary_releases_year}
+              </a>
+            </MenuItem>
+            {window.primary_releases_other.map(release => (
```
Comment (karimkawambwa) [inline dominion/dominion_ui/src/components/ReleaseDropdown.js]: lets try reduce these window variables and assign them to a const. this way we can keep track of them them easily. Also maybe try group some of them like the profile data etc...

File: dominion/dominion_ui/src/components/ReleaseDropdown.js [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+              <MenuItem item className={classes.menuListItem}>
+                <a
+                  href={`${window.location.href.replace(/\/$/, '')}?release=${
+                    release.year
+                  }`}
+                  className={classes.link}
+                >
+                  {release.name} {'  '} {release.year}
```
Comment (karimkawambwa) [inline dominion/dominion_ui/src/components/ReleaseDropdown.js]: you can use caption

**Review by karimkawambwa — COMMENTED** [review]
- Missing data
Screen Shot 2019-03-14 at 11 08 19 AM

- Releases dropdown missing in mobile
Screen Shot 2019-03-14 at 11 12 37 AM

- Increasing dimension causes overflow and dropdown not clickable
Screen Shot 2019-03-14 at 11 13 57 AM

default dimensions 1440x1062 is too close
Screen Shot 2019-03-14 at 11 14 38 AM

- Clicking the country name as folllows
Screen Shot 2019-03-14 at 11 21 53 AM
takes me to `/profiles/country-*` but i think it should not say `County in`
Screen Shot 2019-03-14 at 11 22 06 AM

- the map is off centre of the country and i think we should be able to compare country with country  e.g. Kenya ? 
Screen Shot 2019-03-14 at 11 26 10 AM

- If you type fast and erase, you can get the search results to remain.
Screen Shot 2019-03-14 at 11 26 10 AM

**Comment by KhadijaMahanga** [comment]
> Quick review @KhadijaMahanga, maps are _off_.
> Screenshot from 2019-03-14 10-28-51
> 
> Also in South Africa
@kilemensi  How off? Do you mean drawing other countries?
- I have removed other countries from wazimap geography, if so.

**Comment by kilemensi** [comment]
Off as in the map should be of Kenya and not of Central Africa i.e. should be zoomed and centred on the country in question @KhadijaMahanga.

As in:
Screenshot_2019-03-14 Zeplin - Project

File: dominion/dominion_ui/src/components/Hero/ProfileHero.js (hunk 7 above)
Comment (KhadijaMahanga) [reply dominion/dominion_ui/src/components/Hero/ProfileHero.js:82]: I have used one profile data object

File: dominion/dominion_ui/src/components/Hero/ProfileHero.js (hunk 8 above)
Comment (KhadijaMahanga) [reply dominion/dominion_ui/src/components/Hero/ProfileHero.js]: Okay

**Comment by karimkawambwa** [comment]
LGTM 🚀

**Review by kilemensi — APPROVED** [review]

**Review by KhadijaMahanga — APPROVED** [review]
LGTM

**Pull request merged**